traitlets/config/application.py

```diff
@@ -711,18 +718,31 @@ def _load_config_files(cls, basefilename, path=None, log=None, raise_config_file
         yield each config object in turn.
         """
 
+        def new_loader(name, path):
+            for ext, loader in cls.supported_cfg_loaders.items():
+                if name.endswith(ext):
+                    return loader(name, path, log=log)
+            raise AssertionError("Unknown file-extension in config-file %r!" % name)
```

Comment:
AssertionErrors shouldn't generally be raised. Probably best to raise KeyError or ValueError. It might also be nice to include the supported extensions in the error message.

Reply:
Ok, converted to `ValueError`.

I did the same fix on an another case using `AssertionError` class (missing sub-command).
That's where i copied the code from.  It turned out that it was my code in the first place :-)